pkg/api/types/pagination.go

```diff
@@ -0,0 +1,5 @@
+package types
+
+type PaginatedData[T any] struct {
+	Data []T
```

Comment:
shouldn't this have info on start and end index of the contained data? or page number? some meta data?

Reply:
Yeah it's not complete. I was just messing around with generics. I'd probably flesh this out when doing the list endpoint

Reply:
I would rather add metadata + interface to our current `Lists` objects rather than implementing another model collection

Reply:
And returning a `next` bookmark in the metadata perhaps

Reply:
Such as 
```
type ListMetadata map[string]string
type RecordList struct {
  Metadata ListMetadata
  Items: ?
}
/////
{
  "metadata":{"next":"9e44c0273dc3b838"}
  "items":[
    {...}
  ]
}
```

Reply:
I think the generic is much cleaner

Reply:
Yeah but in this case, we can't really use generics for each entity because we deal with runtime types. 
We sure can define a generic `ObjectList` such as
```
type ObjectList[T:Object] struct{
  Metadata: ListMetadata
  Items: []T
} 
```
and reuse that across

Reply:
Isn't that just what I had? (with the metadata)

Reply:
I'm just seeing a `[]Data` field

Reply:
Oh, lol yeah I see. Yes